### DISCUSSION TITLE
Help uninstalling mkcert

### DISCUSSION THREAD

**Discussion by digicoffey** [FiloSottile/mkcert#592] [Q&A]
Hi all,

I was wondering if someone could help me uninstall `mkcert` properly?

I installed `mkcert` using the following commands:

```
brew install mkcert
mkcert -install
mkcert 0.0.0.0 localhost 127.0.0.1 ::1
```

When I ran these commands, my `/etc/hosts` file had some entries:

Example

```
#127.0.0.1 my.example.com
127.0.0.1 my.second.example.com
```

And I ran my local project like so:

```
npx http-server -p 8030 -S -C cert.pem ./dist
```

I eventually uninstalled `mkcert` like so:

```
brew uninstall mkcert
mkcert -uninstall
```

And since I have done this, I have not been able to access `my.second.example.com` when I run my project locally, ie 

```
yarn start:dev
```

Instead I get the message `This site can't be reached. my.second.example.com refused to connect.`

I've clearly missed a step, presumably around this command `mkcert 0.0.0.0 localhost 127.0.0.1 ::1`, and would appreciate any guidance you could offer.

Thanks very much

**Comment by tigran-sargsyan-w** [comment]
The important thing here is that this command:

```bash
mkcert 0.0.0.0 localhost 127.0.0.1 ::1
```

does not modify `/etc/hosts`, configure DNS, or start a web server.

It only generates a certificate containing those names/IP addresses.

Also:

```text
refused to connect
```

is not normally a certificate-trust error.

A TLS trust problem happens after a TCP connection has been established and usually gives you a certificate/privacy warning.

`ERR_CONNECTION_REFUSED` generally means nothing is listening at the IP/port the browser is trying to connect to.

I would check these separately.

First:

```bash
grep 'my.second.example.com' /etc/hosts
```

You still need something like:

```text
127.0.0.1 my.second.example.com
```

if you rely on `/etc/hosts` for local resolution.

Then verify that your development server is actually listening:

```bash
lsof -nP -iTCP -sTCP:LISTEN
```

or specifically for the old port:

```bash
lsof -nP -iTCP:8030 -sTCP:LISTEN
```

and test it directly:

```bash
curl -v http://127.0.0.1:8030/
```

`mkcert -uninstall` removes the local CA from the trust stores. It does not remove `/etc/hosts` entries and it does not control whether your application listens on a TCP port.

Also note that your old command:

```bash
npx http-server -p 8030 -S -C cert.pem ./dist
```

explicitly started an HTTPS server using the certificate.

If `yarn start:dev` starts a different server, port, bind address or HTTP/HTTPS mode, the old URL may simply no longer point to a listening service.

So I would debug this in this order:

```text
DNS/hosts -> listening port -> HTTP/HTTPS -> certificate trust
```

rather than reinstalling mkcert first.
